### PR TITLE
Simplify IN expression with single element

### PR DESCRIPTION
**Pull request by martint**
Rewrite `x IN (v)` to `x = v`

### CODE REVIEW

File: presto-main/src/test/java/io/prestosql/sql/planner/TestSimplifyIn.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+package io.prestosql.sql.planner;
+
+import io.prestosql.sql.planner.assertions.BasePlanTest;
+import org.testng.annotations.Test;
+
+import static io.prestosql.sql.planner.assertions.PlanMatchPattern.anyTree;
+import static io.prestosql.sql.planner.assertions.PlanMatchPattern.filter;
+import static io.prestosql.sql.planner.assertions.PlanMatchPattern.values;
+
+public class TestSimplifyIn
```
Comment (findepi) [inline presto-main/src/test/java/io/prestosql/sql/planner/TestSimplifyIn.java]: Why not `TestExpressionInterpreter`?

File: presto-main/src/test/java/io/prestosql/sql/planner/TestSimplifyIn.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+{
+    @Test
+    public void testInWithSingleElement()
+    {
+        assertPlan(
+                "SELECT * FROM (VALUES 0) t(a) WHERE a IN (5)",
+                anyTree(
+                        filter("A = 5",
+                                values("A"))));
+
```
Comment (findepi) [inline presto-main/src/test/java/io/prestosql/sql/planner/TestSimplifyIn.java]: blank line (Travis failure)

**Review by findepi — APPROVED** [review]

File: presto-main/src/test/java/io/prestosql/sql/planner/TestSimplifyIn.java (hunk 1 above)
Comment (martint) [reply presto-main/src/test/java/io/prestosql/sql/planner/TestSimplifyIn.java]: I wanted a higher level test that covers it. The logic is now in ExpressionInterpreter, but might move to an independent rule in the future. I'll add it there, too.

File: presto-main/src/test/java/io/prestosql/sql/TestExpressionInterpreter.java [hunk 3]
```diff
@@ -455,6 +455,9 @@ public void testIn()
         assertOptimizedEquals("bound_decimal_short in (123.455, 123.46, 123.45)", "true");
         assertOptimizedEquals("bound_decimal_long in (12345678901234567890.123, 9876543210.9874561204, 9876543210.98745612035)", "true");
         assertOptimizedEquals("bound_decimal_long in (9876543210.9874561204, null, 9876543210.98745612035)", "null");
+
+        assertOptimizedEquals("unbound_integer in (1)", "unbound_integer = 1");
+        assertOptimizedEquals("unbound_integer in (unbound_integer)", "unbound_integer = unbound_integer");
```
Comment (findepi) [inline presto-main/src/test/java/io/prestosql/sql/TestExpressionInterpreter.java]: ideally this should roll up to `unbound_integer IS NOT NULL` (for types other than floating point)

maybe you could also have a test with `var IN (some_other_var)`?

**Review by findepi — APPROVED** [review]

File: presto-main/src/test/java/io/prestosql/sql/TestExpressionInterpreter.java (hunk 3 above)
Comment (findepi) [reply presto-main/src/test/java/io/prestosql/sql/TestExpressionInterpreter.java]: btw `in` should be uppercase

Comment (martint) [reply presto-main/src/test/java/io/prestosql/sql/TestExpressionInterpreter.java]: > ideally this should roll up to unbound_integer IS NOT NULL (for types other than floating point)

I knew you were going to say that :)

Yeah, we should add another optimization for that at some point, but is seems like a pretty uncommon thing to occur in real queries.

Comment (findepi) [reply presto-main/src/test/java/io/prestosql/sql/TestExpressionInterpreter.java]: Right, i don't expect you do add this other optimization now. I'm just saying that, from test perspective, it's slightly better to have two cases (same var, different vars).

**Pull request merged**